metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -303,34 +303,66 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         ).topics
         extra_topic_details = self.fetch_extra_topic_details(topics.keys())
 
-        for t, t_detail in topics.items():
-            self.report.report_topic_scanned(t)
-            if self.source_config.topic_patterns.allowed(t):
+        for topic, topic_detail in topics.items():
+            self.report.report_topic_scanned(topic)
+            if self.source_config.topic_patterns.allowed(topic):
                 try:
                     yield from self._extract_record(
-                        t, t_detail, extra_topic_details.get(t)
+                        topic, "", topic_detail, extra_topic_details.get(topic)
                     )
                 except Exception as e:
-                    logger.warning(f"Failed to extract topic {t}", exc_info=True)
+                    logger.warning(f"Failed to extract topic {topic}", exc_info=True)
                     self.report.report_warning(
-                        "topic", f"Exception while extracting topic {t}: {e}"
+                        "topic", f"Exception while extracting topic {topic}: {e}"
                     )
             else:
-                self.report.report_dropped(t)
+                self.report.report_dropped(topic)
+
+        # Get all subjects from schema registry and ingest them as SCHEMA DatasetSubTypes
+        for subject in self.schema_registry_client.get_subjects():
+            try:
+                yield from self._extract_record(
+                    "", subject, topic_detail=None, extra_topic_config=None
+                )
+            except Exception as e:
+                logger.warning(f"Failed to extract subject {subject}", exc_info=True)
+                self.report.report_warning(
+                    "subject", f"Exception while extracting topic {subject}: {e}"
+                )
 
     def _extract_record(
         self,
         topic: str,
+        subject: str,
```

Comment:
Can we only have either topic or subject and have flag like `is_kafka_entity` or something like that which controls the logic if we are treating the schema as a kafka topic schema or a schema

Reply:
sure updated with only topic and is_subject as boolean to decide if its schema or topic.